docs/content/components/selectmenu.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Select menu
+---
+
+Use select menus to list clickable choices, allow filtering between them, and highlight which ones are selected. See [Primer CSS documentation](https://primer.style/css/components/select-menu) for more details.
+
+## Examples
+
+Use a `DetailsComponent` to toggle the select menu:
+
+```erb
+<%= render Primer::DetailsComponent.new(overlay: :default, reset: true, position: :relative) do |details_component| %>
+  <% details_component.slot(:summary) do %>
+    Choose an option
+  <% end %>
+  <% details_component.slot(:body) do %>
+    <%= render Primer::SelectMenuComponent.new do |menu_component| %>
+      <%= menu_component.slot(:header) do %>
+        My menu
+      <% end %>
+      <%= menu_component.slot(:modal) do %>
+        <button class="SelectMenu-item" role="menuitem">Item 1</button>
+        <button class="SelectMenu-item" role="menuitem">Item 2</button>
+        <button class="SelectMenu-item" role="menuitem">Item 3</button>
+      <% end %>
+    <% end %>
+  <% end %>
+<% end %>
+```
+
+Or make the select menu the `details-menu` element itself:
+
+```erb
+<details class="details-reset details-overlay">
+  <summary class="btn" title="Pick an item">
+    <span>Choose</span>
+    <span class="dropdown-caret"></span>
+  </summary>
+  <%= render Primer::SelectMenuComponent.new(tag: :"details-menu") do |menu_component| %>
```

Comment:
I didn't think the existing DetailsComponent would support using another component for its `body` slot, hence why I just wrote out the HTML for `<details>` and `<summary>` tags here. We could make DetailsComponent's `body` slot no longer required for rendering if `content` is present, where `content` could be `<%= render Primer::SelectMenuComponent.new...`.